MEDMI_Data_Download/met_extraction_script.py

```diff
@@ -67,84 +61,79 @@ def set_region(self, latitude_range, longitude_range):
 
-    def extract_data(self, date_range, latitude_range, longitude_range, outfile_suffix, extra_datasets=[]):
 
-        # Remove main measurement name from extra data list
-        try: extra_datasets.remove(self.MEASUREMENT_NAME)
-        except ValueError: pass
-        except: raise
 
+    def extract_data(self, date_range, latitude_range, longitude_range, outfile_suffix, extract_extra_datasets=False):
         self.set_region(latitude_range, longitude_range)
         self.set_date_range(date_range)
 
-        if len(extra_datasets) > 0:
-            str_extra_data = str(extra_datasets).replace('\'','').replace(' ', '').replace(',', '-')[1:-1]
-            filename = self._filename.format('_extras-{}'.format(str_extra_data), outfile_suffix)
+        if extract_extra_datasets:
+            self._extra_datasets = self.ALLOWED_EXTRA_DATASETS
         else:
-            str_extra_data = '[]'
-            filename = self._filename.format('', outfile_suffix)
-
-        print('extracting {}'.format(self._head_string.format('and extra datasets: {}'.format(str_extra_data), date_range[0], date_range[1])))
-        extraction_dict = self._get_extraction_dict()
-
-        settings = {    'fname': filename,
-                        'headstring': self._head_string.format('and extra datasets: {}'.format(str_extra_data), date_range[0], date_range[1]),
-                        'columnstring': str(self._cols_base + self._cols_specific + extra_datasets)[1:-1] + '\n'}
+            self._extra_datasets = []
 
-        return self._extract_data(extraction_dict, settings, extra_datasets=extra_datasets)
+        print('extracting {}'.format(self._head_string.format(date_range[0], date_range[1])))
+        if extract_extra_datasets:
+            print('... and extra datasets: {}'.format(json.dumps(self._extra_datasets)))
 
-    def extract_data_from_dict(self, extraction_dict, settings, save_to_file=False):
-        return self._extract_data(extraction_dict, settings, save_to_file=save_to_file)
+        extraction_dict = self._get_extraction_dict()
+        settings = self._get_settings(outfile_suffix)
+        return self._extract_data(extraction_dict, settings)
 
 
-    def _extract_data(self, dict, settings, save_to_file=True, extra_datasets=[]):
-        if self._verbose == 1:
+    def _extract_data(self, extraction_dict, settings, save_to_file=True):
+        if self._verbose >= 1:
             print('extracting data for {}'.format(self.MEASUREMENT_NAME))
         elif self._verbose > 1:
-            print('extracting data for {} using dict: {}\n and settings: {}'.format(
-                self.MEASUREMENT_NAME, json.dumps(dict, default=str), json.dumps(settings)))
-        datadata = self._perform_extraction(dict)
+            print('using settings: {}'.format( json.dumps(settings)))
+        datadata = self._perform_extraction(extraction_dict)
+        if save_to_file:
+            self._save_to_file(datadata, settings)
+        return datadata
```

Comment:
With the first logic check being >=, the second test of > is never going to trigger. Should the second test be made a separate if statement?

Reply:
Yes, fixed.